keps/sig-network/4858-ip-cidr-validation/README.md

```diff
@@ -0,0 +1,838 @@
+
+N/A
+
+### Version Skew Strategy
+
+N/A; No interactions between kubernetes components should be affected
+by the change, so skew doesn't matter.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: StrictIPCIDRValidation
+  - Components depending on the feature gate:
+    - kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Yes, it changes the validation of multiple fields.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing surprising.
+
+###### Are there any tests for feature enablement/disablement?
+
+No
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Rollback should not be able to fail.
+
+Rollout would fail if the cluster contains objects with invalid
+IPs/CIDRs that must be re-created as part of the rollout/upgrade. For
+example, if kube-proxy was deployed via a DaemonSet whose template
+contained an invalid IP in `.spec.dnsConfig.nameservers[]`, then it
+would not be possible to recreate the pod after the apiserver was
+updated.
+
+The warnings in the Alpha period will hopefully avert this problem.
+
+###### What specific metrics should inform a rollback?
+
+None; if anything fails, it is likely to involve user workloads. There
+are not any metrics for validation failures (but see the "missing
+metrics" question below).
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+It is automatically in use in any cluster where it is enabled.
+
+###### How can someone using this feature know that it is working for their instance?
+
+"Things didn't break"
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+We could add a metric to kube-apiserver for when a Create/Update gets
+an IP/CIDR-related warning. That might help admins decide whether it
+was safe to upgrade, though it wouldn't especially help them figure
+out _why_. A more useful approach might be to make sure that the
+warnings get logged, with information about specific objects, which
+would be more useful in helping to track down the offending
+users/controllers.
```

Comment:
Maybe a useful metric would be a simpler counter showing the cluster is using bad values.